collector/slave_status.go

```diff
@@ -61,9 +76,17 @@ func ScrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 		for i, col := range slaveCols {
 			if value, ok := parseStatus(*scanArgs[i].(*sql.RawBytes)); ok { // Silently skip unparsable values.
 				ch <- prometheus.MustNewConstMetric(
-					newDesc(slaveStatus, strings.ToLower(col), "Generic metric from SHOW SLAVE STATUS."),
+					prometheus.NewDesc(
+						prometheus.BuildFQName("mysql", slaveStatus, strings.ToLower(col)),
+						"Generic metric from SHOW SLAVE STATUS.",
+						[]string{"master_uuid", "master_host", "channel_name"},
+						nil,
+					),
 					prometheus.UntypedValue,
 					value,
+					columnValue(scanArgs, slaveCols, "Master_UUID"),
```

Comment:
After sleeping on it, I changed my mind. Bringing index variables outside of the for loop is ultimately better engineering, and the loss of readability is very small, since the function is so short.